hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/AbstractJoiner.java

```diff
@@ -313,7 +313,7 @@ private boolean checkMergeTargetIsNotMember(SplitBrainJoinMessage joinMessage) {
 
     private boolean checkClusterStateActiveBeforeMerge(SplitBrainJoinMessage joinMessage) {
         ClusterState clusterState = clusterService.getClusterState();
-        if (clusterState != ClusterState.ACTIVE) {
+        if (!clusterState.isJoinAllowed()) {
```

Comment:
For `membersRemovedInNotActiveState` name, I tried to change it before, but couldn't find a good candidate. There are many methods referring the same name. 
I can change it to `membersRemovedInNotJoinableState`.